### PR TITLE
Albert/bac 134 protocol v11 readable message signing with eip 712

### PR DESCRIPTION
**Pull request by AlbertSu123**

### CODE REVIEW

File: contracts/validations/UserAuthentication.sol [hunk 1]
```diff
@@ -24,16 +25,26 @@ struct MultiSigGroup {
 	uint256 threshold;
 }
 
-contract UserAuthentication is Validation {
+contract UserAuthentication is Validation, EIP712 {
 	using ECDSA for bytes32;
 	bytes32 public constant moduleId = keccak256("UserAuthenticationV1");
-
 	bytes32 public constant ADMIN_ROLE = bytes32("admin");
+	bytes32 public constant USER_OPERATION_TYPEHASH = keccak256(
```
Comment (Chazzzzzzz) [inline contracts/validations/UserAuthentication.sol:32]: why only userAuth???

Comment (AlbertSu123) [reply contracts/validations/UserAuthentication.sol:32]: As opposed to what?

Comment (ken-fun) [reply contracts/validations/UserAuthentication.sol:32]: Should split this ticket to handle other messages -- discussed in person

File: contracts/validations/UserAuthentication.sol [hunk 2]
```diff
@@ -24,16 +25,26 @@ struct MultiSigGroup {
...
+	);
+	string public constant EIP712_NAME = "FunWallet.UserAuthentication";
+	string public constant EIP712_VERSION = "1";
 
 	/**
 	 * @dev This constant is used to define the version of this contract.
 	 */
 	uint256 public constant VERSION = 1;
+	address public immutable ENTRYPOINT;
+	uint256 public immutable CHAINID = block.chainid;
```
Comment (ken-fun) [inline contracts/validations/UserAuthentication.sol:43]: * prefer `CHAIN_ID`

File: test/utils/FunWallet.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ import {
 import { arrayify, BytesLike, defaultAbiCoder, hexlify, Interface, keccak256, parseUnits, toUtf8Bytes } from "ethers/lib/utils";
 import { addressToBytes32, createEncodedSig, EncodedSigStruct, groupSignHash, randomBytesHex } from "../utils/general";
 import { HashZero } from "../aaUtils/testutils";
+import { User } from "aws-sdk/clients/budgets";
```
Comment (ken-fun) [inline test/utils/FunWallet.ts:23]: remove

**Review by ken-fun — APPROVED** [review]

File: contracts/validations/UserAuthentication.sol (hunk 2 above)
Comment (AlbertSu123) [reply contracts/validations/UserAuthentication.sol:43]: fixed!

**Review by Chazzzzzzz — APPROVED** [review]

**Pull request merged**